### ISSUE TITLE
:TernDef doesn't work with files over 250 lines

### ISSUE THREAD

**Issue by Rantanen** [ternjs/tern_for_vim#116]
[Rantanen/node-mumble/lib/MumbleConnection.js#L69](https://github.com/Rantanen/node-mumble/blob/86296a0cd5134b0893aa41a80e69e2e9f2d045b0/lib/MumbleConnection.js#L69)

Cursor on line 69 on top of `_waitForPrefix`. `:TernDef` gives `no definition found`. If the file is cut shorter below 250 lines `:TernDef` works just fine.

250 lines seems to be the limit for whether `tern_for_vim` sends the request with partial or full file content. The partial requests seem to fail at least for most cases where you're looking up function definitions.

The Emacs integration doesn't have the same issue.

**Comment by Rantanen** [comment]
I've got two fixes for this.

- Major increase to the 250 line limit in `tern.py`
- If the buffer isn't modified, read the file from the disk instead of sending it in the request (something that's missing in the `vim` plugin).

Now these two solutions are not mutually exclusive. What I'm curious about is the 250 line limit for partial vs full request. I just removed the limit locally and tested this with a 18 000 lines long file. The results took around 2 seconds independent of whether the full file content was sent in the request or if tern could read it from the disk.

So I would suggest both of these changes:
- Increasing the 250 line limit to around 10 000 (or even removing the limit completely) 
- If the buffer hasn't been modified, let tern read it from the disk.

Do note this applies only to functionality such as :TernDef that is invoked by the user specifically. In this case getting results is way more important than performance considerations.

Docs and completions are different from this. Auto-complete is meant to assist the user and if that takes seconds, it gets in the way - especially in a synchronous environment, which I believe Vim is.
